### DISCUSSION TITLE
Multiple instances?

### DISCUSSION THREAD

**Discussion by dreemur99** [LostRuins/koboldcpp#627] [Q&A]
Is there a way to start different instances?

I have kobold opening and working on something really long that would take a long time to reprocess if I restarted it. Is it possible to keep this instance open, and start a new instance in a different window/port/w.e. and use that instance for something else? preserving the previous instance? If I'm making sense.

I tried duplicating the exe and start it up on its own and while it seems to be up it remains pointing to the same endpoint as the first so I can't actually use it.

**Comment by LostRuins** [comment]
Yes you can do that - but you have to use a different port. So you could run one on the default port 5001, and use `--port 5002` for the second instance. Then you connect to instance 1 with http://localhost:5001 and instance 2 with http://localhost:5002 at the same time.

Bear in mind both will use memory so make sure you have enough.

**Reply by dreemur99** [reply]
Hhhmmm.... alright. Thank you.

**Comment by blueyred** [comment]
One other solution is if you're using a client to access via the API, you can use 
**genkey** to start a seperate generation.

```
genkey - A unique genkey set by the user. When checking a polled-streaming  request, use this key to be able to fetch pending text even if  multiuser is enabled
```
